heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -452,6 +487,28 @@ public static void setTopologyExactlyOnceEnabled(Map<String, Object> conf, boole
     conf.put(Config.TOPOLOGY_EXACTLYONCE_ENABLED, String.valueOf(exactOnce));
   }
 
+  public static void setTopologyPackingAlgorithm(Map<String, Object> conf,
+                                                 PackingAlgorithmType type) {
```

Comment:
Regarding packaging of `IPacking` and `IRepacking`, is the distribution of interfaces between `api` and `spi` packages needed? Given the current structure, moving these interfaces may create confusion. These interfaces are closely linked to other components like `IScheduler`.

We could avoid future movements of interfaces used by users by keeping all the interfaces implemented by custom code in api package?